ui/v2.5/src/components/Galleries/GalleryRecommendationRow.tsx

```diff
@@ -8,15 +8,43 @@ import { getSlickSliderSettings } from "src/core/recommendations";
 interface IProps {
   isTouch: boolean;
   filter: ListFilterModel;
-  result: FindGalleriesQueryResult;
   header: String;
   linkText: String;
+  index: number;
 }
 
 export const GalleryRecommendationRow: FunctionComponent<IProps> = (
   props: IProps
 ) => {
-  const cardCount = props.result.data?.findGalleries.count;
+  const result = useFindGalleries(props.filter);
+  const cardCount = result.data?.findGalleries.count;
+
+  if (result.loading) {
+    return (
+      <div className="recommendation-row gallery-recommendations">
+        <div className="recommendation-row-head">
+          <div>
+            <h2>{props.header}</h2>
+          </div>
+          <a href={`/galleries?${props.filter.makeQueryParameters()}`}>
+            {props.linkText}
+          </a>
+        </div>
+        <Slider
+          {...getSlickSliderSettings(props.filter.itemsPerPage!, props.isTouch)}
+        >
+          {[...Array(props.filter.itemsPerPage)].map((i) => (
+            <div key={i} className="gallery-skeleton skeleton-card"></div>
+          ))}
+        </Slider>
+      </div>
+    );
```

Comment:
Maybe refactor this into a RecommendationRowSkeleton component?

Reply:
I thought about that, but it seemed like overkill since the card skeletons were just CSS on a div.

Reply:
I refactored some of the common code in these row components into a `buildRow`. Feel free to take another look when you get a chance.